### DISCUSSION TITLE
Is there a way to access the value of the overall loss to calculate it while training?

### DISCUSSION THREAD

**Discussion by LucasJSch** [lululxvi/deepxde#1263] [General]
Hello. I want to apply an idea provided in [this paper](https://arxiv.org/pdf/2112.05620.pdf) on how to avoid trivial solutions.

Quoting the paper:
``` To mitigate this, we propose to penalize the gradient of the physical loss. We propose to add a third penalty term to the loss function that penalizes the maximum gradient of the physical loss resulting in an additional term```

I.e. I have to access the value of the loss function (and its past values) to apply a gradient to it. Is there a way to do it with the current state of the code?

**Comment by lululxvi** [comment]
You need to modify the source code.
